ipa-core/src/protocol/hybrid/oprf.rs

```diff
@@ -74,17 +78,82 @@ pub const PRF_CHUNK: usize = 16;
 // multiplications per batch
 const CONV_PROOF_CHUNK: usize = 256;
 
-#[allow(dead_code)]
+#[derive(Default, Debug, Clone, PartialEq)]
+pub struct PRFIndistinguishableHybridReport<BK: SharedValue, V: SharedValue> {
+    prf_of_match_key: u64,
+    value: Replicated<V>,
+    breakdown_key: Replicated<BK>,
+}
+
+impl<BK, V> PRFIndistinguishableHybridReport<BK, V>
+where
+    BK: SharedValue,
+    V: SharedValue,
+{
+    /// ## Panics
+    /// it doesn't. `ShardIndex` is a u32, expanded into a u64 for the mod operation
+    /// `prf_of_match_key % shard_count` will always fit into a `ShardIndex`
+    #[must_use]
+    pub fn shard_picker(&self, shard_count: ShardIndex) -> ShardIndex {
+        let shard_count = u64::from(shard_count);
+        ShardIndex::try_from(self.prf_of_match_key % shard_count)
+            .expect("Modulo a u32 will fit in a u32")
+    }
+}
+
+impl Serializable for PRFIndistinguishableHybridReport<BA8, BA3> {
+    type Size = U12;
+    type DeserializationError = InvalidHybridReportError;
+
+    fn serialize(&self, buf: &mut GenericArray<u8, Self::Size>) {
+        const PRF_MK_SZ: usize = 8;
```

Comment:
nit: `std::mem:size_of_val(self.prf_of_match_key)` - not ideal, but at least won't break badly if it is no longer a u64. Note that it won't be a constant anymore, but it does not seem like a big deal to me

Reply:
that works for `serialize`, but `self` isn't part of the function signature for `deserialize` :/

Reply:
also, if will still break if it's not a u64, or at least if it's bigger, as it won't fit into the `U12` anymore.

Reply:
https://github.com/private-attribution/ipa/blob/f5d9ec786b0750fe73640a45ebd7249462ff1d5f/ipa-core/src/ff/mod.rs#L61

Reply:
I'm not sure I understand the relevance. The complication here is that `prf_of_match_key` is a `u64` in the clear, not any sort of secret sharing that already implements Serializable. 

I refactored this slightly so that it's less repetitive, and left a comment that the implementation relies on it being a `u64`.

Reply:
I am just saying that the size is specified in bytes, not bits